app/models/bookmark.rb

```diff
@@ -1,3 +1,10 @@
+class ControllerValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    controllers = ["dashboard"] + (ActiveRecord::Base.connection.tables.map(&:to_s) + Permission.resources.map {|x| x.tableize }).uniq
```

Comment:
It may be worth putting the table list that @unorthodoxgeek mentioned earlier into a class variable as a simple caching method, which will be more reliable than the Rails.cache (which persists across restarts).